sdk/python/kfp/compiler/v2_compat.py

```diff
@@ -22,24 +23,30 @@
 
 from kubernetes import client as k8s_client
 
-_LAUNCHER_CONTAINER = dsl.UserContainer(name="kfp-launcher",
-                                        image="gcr.io/ml-pipeline/kfp-launcher",
-                                        command="/bin/mount_launcher.sh",
-                                        mirror_volume_mounts=True)
+_DEFAULT_LAUNCHER_IMAGE = "gcr.io/ml-pipeline/kfp-launcher"
 
 
-def update_op(op: dsl.ContainerOp, pipeline_name: dsl.PipelineParam,
-              pipeline_root: dsl.PipelineParam) -> None:
+def update_op(op: dsl.ContainerOp,
+              pipeline_name: dsl.PipelineParam,
+              pipeline_root: dsl.PipelineParam,
+              launcher_image: Optional[str] = None) -> None:
   """Updates the passed in Op for running in v2-compatible mode.
 
     Args:
       op: The Op to update.
       pipeline_spec: The PipelineSpec for the pipeline under which `op`
         runs.
       pipeline_root: The root output directory for pipeline artifacts.
+      launcher_image: An optional launcher image. Useful for tests.
     """
   # Inject the launcher binary and overwrite the entrypoint.
-  op.add_init_container(_LAUNCHER_CONTAINER)
+  image_name = launcher_image or _DEFAULT_LAUNCHER_IMAGE
+  launcher_container = dsl.UserContainer(name="kfp-launcher",
+                                         image=image_name,
+                                         command="/bin/mount_launcher.sh",
```

Comment:
Do we have any documentation regarding the launcher image interface 9e.g. it needs to have the `/bin/mount_launcher.sh` file with certain CLI interface).

Reply:
Not yet. That will come when we've stabilized the feature.